tests/Client/AkeneoTest.php

```diff
@@ -49,6 +50,59 @@ public function it_can_get_products(): void
         $this->assertEquals($product, $response);
     }
 
+    /** @test */
+    public function it_can_upsert_async_products(): void
+    {
+        $product = [
+            'enabled' => true,
+            'family' => 'tshirt',
+            'categories' => ['summer_collection'],
+            'groups' => [],
+            'parent' => null,
+            'values' => [
+                'name' => [
+                    [
+                        'data' => 'top',
+                        'locale' => 'en_US',
+                        'scope' => null,
+                    ],
+                    [
+                        'data' => 'Débardeur',
+                        'locale' => 'fr_FR',
+                        'scope' => null,
+                    ],
+                ],
+                'price' => [
+                    [
+                        'data' => [
+                            [
+                                'amount' => '15.5',
+                                'currency' => 'EUR',
+                            ],
+                            [
+                                'amount' => '15',
+                                'currency' => 'USD',
+                            ],
+                        ],
+                        'locale' => null,
+                        'scope' => null,
+                    ],
+                ],
+            ],
+        ];
+
+        Http::fake([
+            'akeneo/api/rest/v1/products/top' => Http::response($product),
+        ]);
+
+        /** @var Akeneo $akeneo */
+        $akeneo = app(Akeneo::class);
+
+        $response = $akeneo->getProductApi()->upsertAsync('top');
+
+        $this->assertInstanceOf(Promise::class, $response);
```

Comment:
Could you adjust the test so that you actually send the async request, this does not verify that the async request works just that the Akeneo method returns a promise.
Something like this:

```php
        $akeneo->getProductApi()->upsertAsync('code', [
            // data
        ])->then(function($response) {
            // asert response
        })->wait();
```